src/backend/metal/src/command.rs

```diff
@@ -1067,6 +1122,50 @@ impl CommandBuffer {
         self.state.blend_color = Some(*color);
         soft::RenderCommand::SetBlendColor(*color)
     }
+
+    fn push_vs_constants(
+        &mut self, layout_id: native::UniquePipelineLayoutId, buffer_id: Option<u32>
+    ) -> Option<soft::RenderCommand> {
+        self.state.resources_vs.push_constants_buffer_id = buffer_id;
```

Comment:
We should be able to collapse these functions into one and `match` on stage instead

Reply:
I'm a big DRY proponent, and I wasn't able to convince myself that this is a case to merge. Many things are different:
  - `stage` (this is most trivial)
  - `resources_XX` field
  - `soft::XxxCommand` command, which also happens to be the return type

So we could match by stage, but we couldn't return a different type based on that anyway.

Reply:
Hmm yeah fair enough. I was thinking we could pass `resources_XX` as an argument, and perform the push to `commands` directly in these functions (and not return anything at all), but that has some downsides too.